src/library.cpp

```diff
@@ -1155,7 +1155,9 @@ Library read_gds(const char* filename, double unit, double tolerance, const Set<
             case GdsiiRecord::ENDEL:
                 if (polygon) {
                     // Polygons are closed in GDSII (first and last points are the same)
-                    polygon->point_array.count--;
+                    auto& pa = polygon->point_array;
```

Comment:
Looks good, but don't use `auto`. I prefer to be explicit with types:
```c++
Array<Vec2>* pa = &polygon->point_array;
```

Reply:
I understand the point and agree that this change should follow existing style and not introduce any new dependencies (to c++11 features). That said, are you considering using modern c++11 features in newer versions? Memory management seems a burden, and although your array implementation seems great I would opt for std::vector.

Reply:
Not really. I'm not a fan of C++. From my initial tests (the first version of gdstk, local on my computer, started with `std::vector`) the compilation time increases a lot without any benefits to runtime efficiency (actually the opposite in some cases). Furthermore, I find it valuable to do manual memory management: everything is explicit and expensive memory operations are immediately apparent. If I were to do a 3rd rewrite of the library, which is really unlikely, I'd go for a language such as [Zig](https://ziglang.org/), [Odin](https://odin-lang.org/), or, hopefully, [jai](https://github.com/BSVino/JaiPrimer/blob/master/JaiPrimer.md), all of which have manual memory management :-)